addons/account/models/res_partner_bank.py

```diff
@@ -264,6 +273,15 @@ def write(self, vals):
                 # Group initial values by partner_id
                 account_initial_values[account][field] = account[field]
 
+        # Some fields should not be editable based on conditions. It is enforced in the view, but not in python which
+        # leaves them vulnerable to edits via the shell/... So we need to ensure that the user has the rights to edit
+        # these fields when writing too.
+        if ('acc_number' in vals or 'partner_id' in vals) and any(account.lock_trust_fields for account in self):
+            raise UserError(_("You cannot modify the account number or partner of an account that has been trusted."))
+
+        if 'allow_out_payment' in vals and not self.user_has_groups('account.group_validate_bank_account'):
```

Comment:
you should also add this in the create function.
@odony

Reply:
indeed. :+1:

Reply:
@fmdl FYI https://github.com/odoo/odoo/pull/120694